_episodes/04-catservices.md

```diff
@@ -0,0 +1,155 @@
+{: .callout}
+
+> ## The services described here only work in [cms-analysis][cms-analysis]
+> For security reasons, the services described in the following only work if your project is in the `cms-analysis` namespace.
+> You can move the project you have been using so far in this lesson to the `cms-analysis` namespace by going to `Settings` --> `General` --> `Advanced` --> `Transfer project` and select `cms-analysis / CMSDAS / CAT-tutorials` as a new namespace.
+The `cms-analysis / CMSDAS / CAT-tutorials` is to be used for the purpose of testing.
+You should in general select a target namespace in the relevant POG/PAG subgroups.
+{: .callout}
+
+## Using the CAT EOS file service
+CAT has a service account, `cmscat`, that is in the _zh_ group and is a member of the CMS VO. CAT provides a service to request an EOS token in a GitLab CI job to be able to access CMS files on EOS on behalf of the `cmscat` service account.
+
+The service is described in more detail in [here](https://cms-analysis.docs.cern.ch/code/services/#usage-of-gitlab-ci-dataset-service).
+
+The files accessible through this method are hosted in `/eos/cms/store/group/cat`.
+
+> ## The file you need is not there?
+> You can request more datasets to be stored in `/eos/cms/store/group/cat` by creating a MR to [https://gitlab.cern.ch/cms-analysis/services/ci-dataset-files/-/blob/master/datasets.txt](https://gitlab.cern.ch/cms-analysis/services/ci-dataset-files/-/blob/master/datasets.txt).
+{: .callout}
+
+
+
+> ## Exercise: setup a CI job the copies a file using the CAT EOS file service
+>
+> There is a few technical aspects that are involved in this.
+> First, your GitLab CI job needs to be configured to that it creates an authentication token.
+> This is achieved with the following lines:
+> ~~~
+> id_tokens:
+>     MY_JOB_JWT:
+>        aud: "cms-cat-ci-datasets.app.cern.ch"
+> ~~~
+> Second, you need to query a service, hosted at `https://cms-cat-ci-datasets.app.cern.ch`, to give you a short lived token to access a file on EOS, on behalf of the `cmscat` service account.
+This is achieved with the following lines:
+> ~~~
+> XrdSecsssENDORSEMENT=$(curl -H "Authorization: ${MY_JOB_JWT}" "https://cms-cat-ci-datasets.app.cern.ch/api?eospath=${EOSPATH}" | tr -d \")
+> ~~~
+> Where `EOSPATH` is a variable holding a path of a file on EOS.
+> Now you can access the file with a path that includes the newly generated token at the end, as: `root://eoscms.cern.ch/${EOSPATH}?authz=${XrdSecsssENDORSEMENT}&xrd.wantprot=unix`.
+Try copying the file: `/eos/cms/store/group/cat/datasets/MINIAODSIM/RunIISummer20UL17MiniAODv2-106X_mc2017_realistic_v9-v2/DYJetsToLL_M-50_TuneCP5_13TeV-amcatnloFXFX-pythia8/2C5565D7-ADE5-2C40-A0E5-BDFCCF40640E.root`
+{: .challenge}
+
+> ## Solution
+> A possible solution to the exercise above is the following:
+> ~~~
+test_eos_service:
+  image:
+    name: registry.cern.ch/docker.io/cmssw/el7:x86_64
```

Comment:
I guess we should decide whether we want to use the `cmssw` images (which have nothing to do with CMSSW) from O&C or the ones from the DPOA (e.g. `gitlab-registry.cern.ch/cms-cloud/cmssw-docker/cc7-cms:latest`), the latter is used in earlier episodes.

Reply:
The trivial reason for this mismatch is that `xrdcp` is not available in the DPOA image

Reply:
OK, we can figure this out later.